tools/linker/ScanTypeReferenceStep.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using Mono.Cecil;
+using Mono.Linker.Steps;
+using Xamarin.Bundler;
+
+#if MTOUCH
+using ProductException = Xamarin.Bundler.MonoTouchException;
+#else
+using ProductException = Xamarin.Bundler.MonoMacException;
+#endif
+
+namespace Xamarin.Linker.Steps {
+
+	abstract public class ScanTypeReferenceStep : BaseStep {
+		protected readonly List<string> lookfor;
+
+		protected ScanTypeReferenceStep (List<string> list)
+		{
+			lookfor = list;
+		}
+
+		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		{
+			foreach (var module in assembly.Modules) {
+				foreach (var name in lookfor) {
+					if (module.HasTypeReference (name))
+						Report (name, assembly);
+				}
+			}
+		}
+
+		protected abstract void Report (string typeName, AssemblyDefinition assembly);
+	}
+
+	public class PreLinkScanTypeReferenceStep : ScanTypeReferenceStep {
+
+		public PreLinkScanTypeReferenceStep (List<string> list) : base (list)
+		{
+		}
+
+		protected override void Report (string typeName, AssemblyDefinition assembly)
+		{
+			ErrorHelper.Show (new ProductException (1502, false, "One or more reference(s) to type '{0}' already exists inside '{1}' before linking", typeName, assembly));
```

Comment:
The error string belongs in Errors.resx in mmp.csproj. Errors.resx should only be modified in VSMac because VSMac automatically updates Errors.Desiginer.cs. mmp.csproj  should also be built again to update the xlf files.

Reply:
@wachaudh you did not read my commit message

> side note this was first done on d16-6 to ease backports as master
> has multiple changes for localization. A similar PR will be done for
> master once merged.

and, now that I have your attention, where is the process for above documented ?

Reply:
working on it 😄